packages/api/README.md

```diff
@@ -0,0 +1,160 @@
+
+### IPFS Cluster
+
+The nft.storage site talks to IPFS Cluster. You need to run a cluster locally and make it accessible from the internet for development.
+
+Follow the quickstart guide to get an IPFS Cluster up and running: https://cluster.ipfs.io/documentation/quickstart/
+
+Expose the IPFS Proxy by opening the `docker-compose.yml` file and add:
+
+```yaml
+CLUSTER_IPFSPROXY_NODEMULTIADDRESS: /dns4/ipfs0/tcp/5001
+CLUSTER_IPFSPROXY_LISTENMULTIADDRESS: /ip4/0.0.0.0/tcp/9095
+```
+
+...to the `custer0` _environment_ and add:
+
+```yaml
+- '127.0.0.1:9095:9095'
+```
+
+...to the `cluster0` _ports_. Then restart for the changes to take effect.
+
+Install [localtunnel](https://localtunnel.me/) and expose the IPFS Cluster HTTP API and IPFS Proxy API (replacing "USER" with your name):
+
+```
+npm install -g localtunnel
+lt --port 9094 --subdomain USER-cluster-api-nft-storage
+lt --port 9095 --subdomain USER-ipfs-proxy-api-nft-storage
+```
+
+These two URLs should be used for `CLUSTER_API_URL` and `CLUSTER_IPFS_PROXY_API_URL` in the `wrangler.toml` (see below).
+
+There is an npm script you can use to quickly establish these tunnels during development:
+
+```sh
+yarn lt
+```
+
+### Cloudflare Workers initial setup:
+
+> This only needs to be run once when setting up from scratch.
+
+#### Development Setup
+
+Open `wrangler.toml` and add an env for yourself (replacing "USER" with your name and "CF_ACCOUNT" with your Cloudflare account):
+
+```toml
+[env.USER]
+type = "webpack"
+name = "nft-storage-USER"
+account_id = "CF_ACCOUNT"
+workers_dev = true
+route = ""
+zone_id = ""
+vars = { ENV = "dev", DEBUG = "*", CLUSTER_API_URL = "", CLUSTER_IPFS_PROXY_API_URL = "" }
+kv_namespaces = []
+```
+
+Additionally, fill in the `CLUSTER_API_URL` and `CLUSTER_IPFS_PROXY_API_URL` with the localtunnel URLs you obtained when setting up the IPFS Cluster.
+
+```bash
+cd packages/api
+yarn install
+# dev and preview KVs
+wrangler kv:namespace create USERS --preview --env USER
+# cli output something like: `{ binding = "USERS", preview_id = "7e441603d1bc4d5a87f6cecb959018e4" }`
+# but you need to put `{ binding = "USERS", preview_id = "7e441603d1bc4d5a87f6cecb959018e4", id = "7e441603d1bc4d5a87f6cecb959018e4" }` inside the `kv_namespaces`.
+wrangler kv:namespace create NFTS --preview --env USER
+# same as above
+wrangler kv:namespace create NFTS_IDX --preview --env USER
+# same as above
+wrangler kv:namespace create DEALS --preview --env USER
+# same as above
+wrangler kv:namespace create METRICS --preview --env USER
+# same as above
+wrangler kv:namespace create PINS --preview --env USER
+# same as above
+wrangler kv:namespace create FOLLOWUPS --preview --env USER
+# same as above
+wrangler kv:namespace create PINATA_QUEUE --preview --env USER
+# same as above
+```
+
+Go to `/packages/api/src/constants.js` _uncomment_ the first line and run `wrangler publish --env USER`.
```

Comment:
In this step, I am getting:

```
Error: Something went wrong! Status: 400 Bad Request, Details {
  "result": null,
  "success": false,
  "errors": [
    {
      "code": 10021,
      "message": "Uncaught ReferenceError: DEALS is not defined\n  at line 50606 in ./src/constants.js\n  at line 20 in __webpack_require__\n  at line 53167 in ./src/utils/debug.js\n  at line 20 in __webpack_require__\n  at line 54179\n  at line 54356 in ./src/utils/router.js\n  at line 20 in __webpack_require__\n  at line 50799 in ./src/index.js\n  at line 20 in __webpack_require__\n  at line 84\n"
    }
  ],
  "messages": []
}
```

I did the steps above to add the Deals (with success), but it does not feel it got applied. If I comment line 3 from constants, I get the same error for "USERS". How is this loading the references?

Reply:
do you have the deals kv created in your CF account ?

Reply:
After syncing with @alanshaw I could move forward on this one. Two things here:

1. create previous step where we add to the `wrangler.toml` file the kv_namespaces, or make it more clear on the comments above
2. constants.js file uncommented line also needs `SENTRY_DSN`

Reply:
> do you have the deals kv created in your CF account ?

I was missing the `kv_namespaces`

Reply:
the cli doesnt output instructions to add kv namespaces to the toml file ?

Reply:
Both me and @olizilla did not get initially that `preview_id` and `id` should be the same here